pkg/revdial/const.go

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package revdial
+
+// define the constants used to build the URL
+// The listener connects to an user with path [host:port/base]/revdial?id=[id]
+// The dialer listens on the urls:
+// [host:port/base]/revdial for the reverse connections
+// [host:port/base]/proxy/[id]/[path] for the reverse proxied to [path]
```

Comment:
this is too static I think. We need a more flexible way to map these path to IDs, i.e. dynamic in code, not with fixed paths in some mux.

E.g. a request comes in for `/clusters/root:org:ws/api/v1/namespaces/default/pods/foo/logs`. We have to lookup the pod and check where it is scheduled to via `state.workload.kcp.dev/<synctarget-name>` label (cc @jmprusi). `<synctarget-name>` (this will be either a UID or a hash, see https://docs.google.com/document/d/1RyyIuOq20PX6kAGVuzCvXdqpipsnci5E1c887Y5GKbg/edit#), but should equal the revdial id. Hence, when we have that id from the pod's label, we pick the right reverse proxy.

At this point, the request is already authorized by kcp. But it would be good to have some more authorization (= filtering of paths) on the syncer side. Note that the syncer today is pretty privileged.

Reply:
Note: we don't have pods yet in the system. We are looking into inverse syncing which will make pods replicated from downstream back to kcp. Hence, for the time being we could pick `state.workload.kcp.dev/<synctarget-name>` and with that the id from the namespace instead of the pod.